internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -426,6 +426,49 @@ func (r *BMACReconciler) reconcileAgentInventory(bmh *bmh_v1alpha1.BareMetalHost
 
 }
 
+// Utility to verify whether a BMH should be reconciled based on the InfraEnv
+//
+// This function verifies three things:
+//
+// 1. InfraEnv existsD
+// 2. InfraEnv's ISODownloadURL exists
+// 3. The BMH.Spec.URL is the same to the InfraEnv's ISODownload URL
+//
+// If the three checks above are true, then the reconcile won't happen
+// and the reconcileBMH function will return.
```

Comment:
A bit connected to https://github.com/openshift/assisted-service/pull/1943... How about checking already here if BMH is detached ? If that would be the case, you would spare an unnecessary call to `reconcileBMH()` (when #1943 lands in, this check will happen there anyway). It adds some logic for a potential (but not sure how big) performance gain in big environments. Feel free to drop this idea if you see it as unnecessary complication, though :)

Reply:
Good idea! 

I would rather do a refactor in a follow-up PR (if you agree) once both PR land so that we can unify the logic. FWIW, these PRs were created to address separate issues, hence my interest in keeping them isolated if possible.

Reply:
Sure, let's get this one as a "feature" and the follow-up as a "performance optimization"